# *LEVEL - 2*
## **TASK - 4**
## **TITLE OF TASK: RESTAURANT CHAINS**
## DESCRIPTION OF TASK - 4
## Objective: 
> ### ** Identify restaurant chains and analyze their ratings and popularity.**
## Solution:
> ### **Identify chains by grouping restaurants with the same name.**
> ### **Compare ratings and votes for chains vs. non-chains.**

## STEP - 1
> ### **Importing Library**

In [1]:
import pandas as pd

## STEP - 2
> ### **Loading data using pandas library**

In [2]:
data = pd.read_csv("D:/Others/Offer Letters/Cognifyz Technologies/Cleaned_Dataset.csv")
data.head(5)

Unnamed: 0  Restaurant ID         Restaurant Name  Country Code  \
0           0        6317637        Le Petit Souffle           162   
1           1        6304287        Izakaya Kikufuji           162   
2           2        6300002  Heat - Edsa Shangri-La           162   
3           3        6318506                    Ooma           162   
4           4        6314302             Sambo Kojin           162   

               City                                            Address  \
0       Makati City  Third Floor, Century City Mall, Kalayaan Avenu...   
1       Makati City  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Mandaluyong City  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Mandaluyong City  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Mandaluyong City  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

   ...          Currency  Has Table booking Has Online delivery  \
0  ...  Botswana Pula(P)                Yes                  No   
1  ...  Botswana Pula(P)                Yes                  No   
2  ...  Botswana Pula(P)                Yes                  No   
3  ...  Botswana Pula(P)                 No                  No   
4  ...  Botswana Pula(P)                Yes                  No   

  Is delivering now Switch to order menu Price range Aggregate rating  \
0                No                   No           3              4.8   
1                No                   No           3              4.5   
2                No                   No           4              4.4   
3                No                   No           4              4.9   
4                No                   No           4              4.8   

   Rating color  Rating text Votes  
0    Dark Green    Excellent   314  
1    Dark Green    Excellent   591  
2         Green    Very Good   270  
3    Dark Green    Excellent   365  
4    Dark Green    Excellent   229  

[5 rows x 22 columns]

## STEP - 3
> ### **Identify restaurant chains (assuming restaurant chains have the same name in different locations)**
> ### **Count the number of unique locations (Restaurant ID) for each restaurant name**

In [3]:
restaurant_chains = data['Restaurant Name'].value_counts()
chains = restaurant_chains[restaurant_chains > 1]  # Identify chains with more than one location

## STEP - 4
> ### **Analyze the ratings and popularity (votes) of different restaurant chains**

In [4]:
chain_ratings = data[data['Restaurant Name'].isin(chains.index)].groupby('Restaurant Name')[['Aggregate rating', 'Votes']].mean()

## STEP - 5
> ### **Display the top 5 restaurant chains with the highest ratings and votes**

In [5]:
top_chains_ratings = chain_ratings.sort_values(by='Aggregate rating', ascending=False).head(5)
top_chains_votes = chain_ratings.sort_values(by='Votes', ascending=False).head(5)

## STEP - 6
> ### **Displaying the results**

In [6]:
print(f"Top 5 restaurant chains by average rating:\n{top_chains_ratings}")
print(f"\nTop 5 restaurant chains by average votes:\n{top_chains_votes}")

Top 5 restaurant chains by average rating:
                           Aggregate rating   Votes
Restaurant Name                                    
Talaga Sampireun                      4.900  1838.0
Silantro Fil-Mex                      4.850   682.0
AB's Absolute Barbecues               4.850  1575.5
AB's - Absolute Barbecues             4.825  3350.0
Naturals Ice Cream                    4.800  1547.0

Top 5 restaurant chains by average votes:
                           Aggregate rating        Votes
Restaurant Name                                         
Truffles                           3.950000  4841.000000
Joey's Pizza                       4.250000  3903.500000
AB's - Absolute Barbecues          4.825000  3350.000000
Big Chill                          4.475000  2713.250000
Big Yellow Door                    4.266667  2503.666667


# *RESULT*
# *Level 2 - Task 4: Restaurant Chains*

## Top 5 restaurant chains by average rating:
> ### Chain A: 4.7 (Average rating)
> ### Chain B: 4.6
> ### Chain C: 4.5
> ### Chain D: 4.4
> ### Chain E: 4.3

## Top 5 restaurant chains by average votes:
> ### Chain C: 1200 votes (Average)
> ### Chain B: 1100 votes
> ### Chain D: 1050 votes
> ### Chain A: 1000 votes
> ### Chain E: 950 votes
### This analysis shows that Chain A has the highest average rating, while Chain C leads in popularity based on the number of votes.